File: MatriciKalk_java/Struttura/Struttura.java
package Struttura;

import Exception.Eccezione;
import java.math.*;
import java.util.Vector;

public abstract class Struttura {
    private Vector<Integer> dim=new Vector<Integer>();
    private int id;
    
    public Struttura(int[] vet, int n){
        for(int i=0; i<n; i++)
            dim.add(vet[i]);
    }
    
    public Struttura(int d1, int d2){
        dim.add(d1);
        dim.add(d2);
    }
     public int getDimAt(int p){
         return dim.elementAt(p);
     }
     
     public int getId(){
         return id;
     }
    
     public abstract Struttura somma(Struttura mat)throws Eccezione;
     public abstract Struttura differenza(Struttura mat)throws Eccezione;
     public abstract Struttura moltiplicazione(Struttura mat)throws Eccezione;
     public abstract Struttura potenza(int exp);
     
    protected void setId(int i){
        id=i;
    }
}

File: MatriciKalk_c++/struttura.cpp
#include "struttura.h"

Struttura::Struttura(int* d, int numDim){
    for(int i=0; i<numDim; i++){
         dim.append(*d);
         d++;
    }
}


Struttura::~Struttura(){
    QVector<int>::iterator it=dim.begin();
    for(;it!=dim.end();++it){
        it=dim.erase(it);
        it--;
    }
}

Struttura::Struttura(int d1, int d2){
    dim.append(d1);
    dim.append(d2);
}

int Struttura::getDimAt(int i) const{
    return dim.at(i);
}

void Struttura::setDimAt(int pos, int v){
    dim[pos]=v;
}

void Struttura::setId(int i) {
    id=i;
}

int Struttura::getId()const{
    return id;
}
